dataset.py

```diff
@@ -1224,142 +1224,30 @@ def evaluate_scores(self, scores):
-        else:
-            image = [Image.open(image_path).convert('RGB') for image_path in image_paths]
-            image = [self.image_preprocess(img) for img in image]
-        
-        texts = [str(item['prompt'])]
-        item = {"images": image, "texts": texts}
-        
-        return item
-    
-    def correlation(self, our_scores, human_scores):
-        pearson = calc_pearson(human_scores, our_scores)
-        print(f"Pearson's Correlation (no grouping): ", pearson)
-        
-        kendall_b = calc_metric(human_scores, our_scores, variant="tau_b")
-        print(f'Kendall Tau-B Score (no grouping): ', kendall_b)
-        
-        pairwise_acc = calc_metric(human_scores, our_scores, variant="pairwise_acc_with_tie_optimization")
-        print(f'Pairwise Accuracy Score (no grouping): ', pairwise_acc)
-        
-        results = {
-            'pearson': pearson,
-            'kendall_b': kendall_b,
-            'pairwise_acc': pairwise_acc,
-        }
-        return results
-    
-    def evaluate_scores(self, scores, indices=list(range(527))):
-        scores_i2t = scores
-        human_avg_scores_alignment = [np.array(self.images[idx]['human_alignment']).mean() for idx in range(len(self.images))]
-        our_scores = scores_i2t.mean(axis=1)
-        our_scores = [float(our_scores[idx][0]) for idx in range(len(self.images))]
-        alignment_correlation = self.correlation(our_scores, human_avg_scores_alignment)
-        results = {
-            'alignment': alignment_correlation,
-        }
-        return results
-
-
 class GenAIBench_Image(Dataset):
-    # GenAIBench with 527 prompts x 6 images (from 6 models)
+    # GenAIBench with 527/1600 prompts x 6 images (from 6 models)
     def __init__(self,
                  image_preprocess=None,
                  root_dir="./",
                  download=True,
+                 num_prompts=1600, # num_prompts must be 527(VQAScore paper) or 1600(GenAI-Bench paper)
                  return_image_paths=True):
-        self.root_dir = os.path.join(root_dir, 'GenAI-Image-527')
+        
+        self.root_dir = os.path.join(root_dir, f'GenAI-Image-{num_prompts}')
         self.models = ['DALLE_3', 'SDXL_Turbo', 'DeepFloyd_I_XL_v1', 'Midjourney_6', 'SDXL_2_1', 'SDXL_Base']
         
         self.image_preprocess = image_preprocess
         self.return_image_paths = return_image_paths
         if self.return_image_paths:
             assert self.image_preprocess is None, "Cannot return image paths and apply transforms"
-        
-        self.download_links = {
-            model_name: f"https://huggingface.co/datasets/zhiqiulin/GenAI-Bench-527/resolve/main/{model_name}.zip" for model_name in self.models
-        }
+        if num_prompts == 527:
+            self.download_links = {
+                model_name: f"https://huggingface.co/datasets/zhiqiulin/GenAI-Bench-527/resolve/main/{model_name}.zip" for model_name in self.models
+            }
+        else:
```

Comment:
This is not robust. What if someone inputs 800? If we only support 527 and 1600, then we should use assert to check no other options are passed in